### PR TITLE
Refactor/unnamed semaphores,drbitboy fixes,plus catch2

### PR DESCRIPTION
**Pull request by drbitboy**

### CODE REVIEW

File: tests/CMakeLists.txt.gtest.obsolete
Comment (a-sevin) [inline tests/CMakeLists.txt.gtest.obsolete:1]: I think you can remove it

**Comment by DasVinch** [comment]
A little out of my depth here...
Assuming my understanding is correct, simple question: should `catch.hpp` be commited at all, since it's a/ 17k lines and b/ automatically generated?

Other than that, I'm giving all confidence to @drbitboy  :)

**Comment by drbitboy** [comment]
> A little out of my depth here... Assuming my understanding is correct, simple question: should `catch.hpp` be commited at all, since it's a/ 17k lines and b/ automatically generated?
> 
> Other than that, I'm giving all confidence to @drbitboy :)

yes, I should comment what I am doing there.  that is the simplest way to use catch:  https://android.googlesource.com/platform/external/catch2/+/HEAD/docs/tutorial.md#getting-catch2

File: tests/CMakeLists.txt.gtest.obsolete
Comment (drbitboy) [reply tests/CMakeLists.txt.gtest.obsolete:1]: done.

**Pull request merged**